skytap/client.go

```diff
@@ -188,40 +212,91 @@ func (c *Client) newRequest(ctx context.Context, method, path string, body inter
 	return req, nil
 }
 
-func (c *Client) do(ctx context.Context, req *http.Request, v interface{}) (*http.Response, error) {
-	var resp *http.Response
-	var err error
-	var makeRequest = true
+func (c *Client) do(ctx context.Context, req *http.Request, v interface{}, state *environmentVMRunState, payload responseComparator) (*http.Response, error) {
```

Comment:
Probably wrap the client in top of a library as [go-retriable](https://godoc.org/github.com/hashicorp/go-retryablehttp#Client) Instead of self implement the logic, it will provide exponential backoff and jitter.

I will probably not differentiate the request by `verb` in do an let the implementation of the check Resource to take that decision.

Reply:
Going to create a new ticket to implement this